subsys/prefetch/src/main/java/org/commonjava/indy/subsys/prefetch/PrefetchFrontier.java

```diff
@@ -105,20 +111,56 @@ public void scheduleRepo( final RemoteRepository repo, final List<String> paths
         }
     }
 
-    public Map<RemoteRepository, List<ConcreteResource>> remove( final int size )
+    public void rescheduleForRescan()
+    {
+        if ( shouldSchedule && !hasMore )
+        {
+            lockAnd( t -> {
+                for ( RemoteRepository repo : repoQueue )
+                {
+                    if ( repo.isPrefetchRescan() )
+                    {
+                        String rescanTime = repo.getPrefetchRescanTimestamp();
+                        if ( StringUtils.isBlank( rescanTime ) || isNowAfter( rescanTime ) )
+                        {
+                            repo.setPrefetchRescanTimestamp(
```

Comment:
Do we have a call to store this updated copy of the repo again? Maybe I'm just missing it? If we don't, and we add it, we'll need to be sure to avoid a recursive loop (triggering the listener method for the prefetch scheduler).

Reply:
Oh, forgot to store it. Will add it with a no event fire flag.